sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala

```diff
@@ -191,22 +191,19 @@ class ResolveCatalogs(val catalogManager: CatalogManager)
           s"because view support in catalog has not been implemented yet")
 
     case c @ CreateNamespaceStatement(NonSessionCatalogAndNamespace(catalog, ns), _, _) =>
-      CreateNamespace(catalog.asNamespaceCatalog, ns, c.ifNotExists, c.properties)
+      CreateNamespace(catalog, ns, c.ifNotExists, c.properties)
 
     case DropNamespaceStatement(NonSessionCatalogAndNamespace(catalog, ns), ifExists, cascade) =>
       DropNamespace(catalog, ns, ifExists, cascade)
 
     case DescribeNamespaceStatement(NonSessionCatalogAndNamespace(catalog, ns), extended) =>
-      DescribeNamespace(catalog.asNamespaceCatalog, ns, extended)
+      DescribeNamespace(catalog, ns, extended)
```

Comment:
do we still need the separation between `DescribeNamespaceStatement` and `DescribeNamespace`?

Reply:
sorry, could you elaborate on your question?

Reply:
It's simpler to
1. create `DescribeNamespace(UnresolvedNamespace(...))` in parser
2. match `DescribeNamespace(ResolvedNamespace(catalog, ident))` in planner and convert it to physical node.

Reply:
But we still need to handle the case where catalog is a session catalog (and I don't think we want to handle this case in planner).

Reply:
yea, we should have an extra analyzer rule to catch
```
case DescribeNamespace(ResolvedNamespace(catalog, ident)) if isSessionCatalog(catalog)
```
and convert to v1 command.

Reply:
We can do this for the following commands right?
```
CreateNamespaceStatement
DropNamespaceStatement
DescribeNamespaceStatement
ShowNamespacesStatement
ShowTablesStatement
AlterNamespaceSetPropertiesStatement
AlterNamespaceSetLocationStatement
```
In that case, can I do it as a follow up PR?

Reply:
sounds good